CellFabric/cell_fabric/canvas.py

```diff
@@ -63,10 +64,16 @@ def addWireAndMultiViaSet( self, netName, pinName, wire, c, listOfPairs, *, bIdx
         mnP = min(tuples)
         mxP = max(tuples)
 
-# should be the real enclosure but this finds the next grid point
-        enclosure = 1
-        (mn, _) = wire.spg.inverseBounds( mnP-enclosure)
-        (_, mx) = wire.spg.inverseBounds( mxP+enclosure)
+        # Make sure wire is using EnclosureGrid for spg
+        assert isinstance(wire.spg, EnclosureGrid)
+        # Find nearest spg grid coordinate
+        # whilte adjusting for viaWidth //2 + viaEncA
+        enclosure = wire.spg.grid[1][0] - wire.spg.grid[0][0]
```

Comment:
This is okay for now but it really isn't the right thing. We should store the enclosure values in the via and use that instead of computing from the stop point grid which might we wrong. Also, we should be able to use a centered grid or a custom grid as well, not just EnclosureGrid

Reply:
Made a preliminary commit addressing this.